src/components/CatCatalogue.js

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useEffect } from "react";
+
+import CatService from "../services/CatServices";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faTrashAlt } from "@fortawesome/free-solid-svg-icons";
+const PAGE_SIZE = 10;
+
+function CatCatalogue() {
+  const [cats, setCats] = useState([]);
+  const [pageIndex, setPageIndex] = useState(0);
+  const [hasNextPage, setHasNextPage] = useState(false);
+  const [catName, setCatName] = useState("");
+  const [responseMessage, setResponseMessage] = useState("");
+
+  const getCats = (pageIndex) => {
+    CatService.getAllCats({
+      pageNumber: pageIndex,
+      pageSize: PAGE_SIZE,
+      sortBy: "id",
+    }).then((response) => {
+      const cats = response.results;
+
+      setCats(cats);
+      setPageIndex(pageIndex);
+      setHasNextPage(response.metadata.hasNextPage);
+    });
+  };
+
+  useEffect(() => {
+    const pageIndex = 1;
+    getCats(pageIndex);
+  }, []);
+
+  const moveToNextPage = () => {
+    getCats(pageIndex + 1);
+  };
+
+  const moveToPrevPage = () => {
+    getCats(pageIndex - 1);
+  };
+
+  const addCat = () => {
+    CatService.addNewCat(catName).then((response) => {
+      if (response.ctime != undefined) {
+        setResponseMessage("Cat added!");
+      } else {
+        setResponseMessage("Error while adding Cat.");
+      }
```

Comment:
[question] Have you tested this flow? I am not sure if it works like you think it work.
In what case will `ctime` be `undefined`? In such case, would code even come here?
(To answer the second part, please refer [to here](https://github.com/axios/axios#handling-errors))
Also I recommend using `async` and `await` instead of `.then().then().then().catch()..` etc